objects/Fan-MadeScenariosCampaignsMiscellany.66e97c.ttslua

```diff
@@ -0,0 +1,504 @@
+-- Utility memory bag by Directsun
+-- Version 2.5.2
+-- Fork of Memory Bag 2.0 by MrStump
+
+function updateSave()
+    local data_to_save = {["ml"]=memoryList}
+    saved_data = JSON.encode(data_to_save)
+    self.script_state = saved_data
+end
+
+function combineMemoryFromBagsWithin()
+  local bagObjList = self.getObjects()
+  for _, bagObj in ipairs(bagObjList) do
+    local data = bagObj.lua_script_state
+      if data ~= nil then
+        local j = JSON.decode(data)
+        if j ~= nil and j.ml ~= nil then
+          for guid, entry in pairs(j.ml) do
+            memoryList[guid] = entry
+          end
+        end
+      end
+    end
+end
+
+function updateMemoryWithMoves()
+    memoryList = memoryListBackup
+    --get the first transposed object's coordinates
+    local obj = getObjectFromGUID(moveGuid)
+
+    -- p1 is where needs to go, p2 is where it was
+    local refObjPos = memoryList[moveGuid].pos
+    local deltaPos = findOffsetDistance(obj.getPosition(), refObjPos, nil)
+    local movedRotation = obj.getRotation()
+    for guid, entry in pairs(memoryList) do
+        memoryList[guid].pos.x = entry.pos.x - deltaPos.x
+        memoryList[guid].pos.y = entry.pos.y - deltaPos.y
+        memoryList[guid].pos.z = entry.pos.z - deltaPos.z
+        -- memoryList[guid].rot.x = movedRotation.x
+        -- memoryList[guid].rot.y = movedRotation.y
+        -- memoryList[guid].rot.z = movedRotation.z
+    end
+
+    --theList[obj.getGUID()] = {
```

Comment:
Any reason not to clean this up?

Reply:
Code is copy&paste from the official scenario box to the left of it.

If I were to look into it, I'd probably remove a lot of code from it 😅